addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -6245,6 +6261,111 @@ registry.ImageTools = ImageHandlerOption.extend({
     async setImgShapeRotateRight(previewMode, widgetValue, params) {
         await this._setImgShapeRotate(90);
     },
+    /**
+     * Set the hover effects of the image shape.
+     *
+     * @see this.selectClass for parameters
+     */
+    async setImgShapeHoverEffect(previewMode, widgetValue, params) {
+        const img = this._getImg();
+        delete img.dataset.hoverEffectColor;
+        delete img.dataset.hoverEffectIntensity;
+        delete img.dataset.hoverEffectStrokeWidth;
+        if (params.name === "hover_effect_overlay_opt") {
+            img.dataset.hoverEffectColor = this._getCSSColorValue("black-25");
+        } else if (params.name === "hover_effect_outline_opt") {
+            img.dataset.hoverEffectColor = this._getCSSColorValue("primary");
+            img.dataset.hoverEffectStrokeWidth = 10;
+        } else {
+            img.dataset.hoverEffectIntensity = 20;
+            if (params.name !== "hover_effect_mirror_blur_opt") {
+                img.dataset.hoverEffectColor = "rgba(0, 0, 0, 0)";
+            }
+        }
+        await this._applyOptions();
+    },
+    /**
+     * @see this.selectClass for parameters
+     */
+    async selectDataAttribute(previewMode, widgetValue, params) {
+        await this._super(...arguments);
+        if (["hoverEffectIntensity", "hoverEffectStrokeWidth"].includes(params.attributeName)) {
+            await this._applyOptions();
+        }
+    },
+    /**
+     * Set the color of hover effects.
+     *
+     * @see this.selectClass for parameters
+     */
+    async setHoverEffectColor(previewMode, widgetValue, params) {
+        const img = this._getImg();
+        let defaultColor = "rgba(0, 0, 0, 0)";
+        if (img.dataset.hoverEffect === "overlay") {
+            defaultColor = "black-25";
+        } else if (img.dataset.hoverEffect === "outline") {
+            defaultColor = "primary";
+        }
+        img.dataset.hoverEffectColor = this._getCSSColorValue(widgetValue || defaultColor);
+        await this._applyOptions();
+    },
```

Comment:
What are the options doing in web_editor for an option that can only work in website ? :thinking: 

Note: I tried to launch mass_mailing, clicked on an image and got a traceback. Probably the same thing. Probably none of this PR should target web_editor? The shapes are not in mass_mailing either, right?

Reply:
> What are the options doing in web_editor for an option that can only work in website ? 🤔
> 
> Note: I tried to launch mass_mailing, clicked on an image and got a traceback. Probably the same thing. Probably none of this PR should target web_editor? The shapes are not in mass_mailing either, right?

Actually, the traceback is due to one of my recent changes. It happened because I used `getParent()` on an element that wasn't present in `mass_mailing` to transfer the hover effects to `website animate`. It's fixed now.

As we discussed, we'll keep the JavaScript code in `web_editor` for the time being because it's closely tied to all the shapes' code that's also used in `mass_mailing`. I will relocate XML options and anything else feasible to Website.